src/components/BuilderRoute.tsx

```diff
@@ -9,26 +9,30 @@ import { useCurrentNodeContext } from './CurrentNodeProvider';
 const BuilderRoute: FC = () => {
   const { id, nodeId } = useParams();
   const { pathways } = usePathwaysContext();
-  const { setPathway } = useCurrentPathwayContext();
+  const { pathwayRef, setPathway, resetPathway } = useCurrentPathwayContext();
   const { setCurrentNode } = useCurrentNodeContext();
   const pathwayId = decodeURIComponent(id);
   const pathwayIndex = useMemo(() => pathways.findIndex(pathway => pathway.id === pathwayId), [
     pathwayId,
     pathways
   ]);
   const pathway = pathways[pathwayIndex];
-  const currentNode = pathway?.nodes?.[decodeURIComponent(nodeId)];
+  const currentNodeId = decodeURIComponent(nodeId);
 
   useEffect(() => {
-    setPathway(pathway);
-  }, [pathway, setPathway]);
+    if (pathway && pathwayRef?.current?.id === pathway.id) setPathway(pathway);
+    else if (pathway) resetPathway(pathway);
+  }, [pathway, pathwayRef, setCurrentNode, setPathway, resetPathway]); // eslint-disable-line
```

Comment:
git won't let me insert a suggestion but I was able to lint fine without the magic `// esline-disable-line` comment.  Are you getting an error without it or can we remove this?

Reply:
This should be fine now. It was failing before but I fixed the error and forgot to remove the comment